presto-docs/src/main/sphinx/functions/bitwise.rst

```diff
@@ -28,4 +28,27 @@ Bitwise Functions
 
     Returns the bitwise XOR of ``x`` and ``y`` in 2's complement representation.
 
+.. function:: bitwise_shift_left_logical(x, shift, bits) -> bigint
+
+    Shift left logical operation on ``x`` (treated as ``bits``-bit signed integer)
+    shifted by ``shift`` in 2's complement representation.
+
+        SELECT bitwise_shift_left_logical(7, 2, 4); -- 12
```

Comment:
12 is a 5-bit signed number. This doesn't make sense to me. if 12 is a 4-bit signed number, then we cannot represent a 64-bit signed number with bigint.

Reply:
(not necessarily that operation doesn't make sense, maybe just the description doesn't make sense).

Reply:
Overall, for logical left shift, you probably will always fill 0, so `bits` is how many bits to keep after shift (where to truncate). So describing the number as n-bit signed just adds confusion.

For logical right shift, what you want to describe is what to fill with (0 or 1), so maybe defining the sign-bit makes sense. Though I think it makes more sense to say how many bits I want to operate on, and whether I'd fill it with 0 or 1 (`logical_right_shift(number, shift, bits, 0/1)`) Describing logical shift with signs is just very confusing to me.

Reply:
If right shift is also going to fill with 0, then we should not describe the function using "signed number" at all.

Reply:
By definition, logical right shift will always fill in 0s, so it does not preserve the sign of the input. Meanwhile, arithmetic right shift will fill with the MSB of the input. This operation will preserve the sign of the input. Therefore, I think we should just stick to these default conventions. I feel like adding the `0/1` field as a 4th input would just make it more confusing.

Reply:
I did remove the "signed number" part in the description though, I can see how that can be confusing.